### ISSUE TITLE
Generator fails with identical view class names in different files

### ISSUE THREAD

**Issue by tristanlabelle** [AvaloniaUI/Avalonia.NameGenerator#92]
I'm using a slightly unusual file organization for my UI with:

```
FooPage/
  View.axaml
  ViewModel.cs
BarPage/
  View.axaml
  ViewModel.cs
```

However, this fails to compile with:

```
Error	AXN0002	System.ArgumentException: The hintName 'View.g.cs' of the added source file must be unique within a generator.
Parameter name: hintName
   at Microsoft.CodeAnalysis.AdditionalSourcesCollection.Add(String hintName, SourceText source)
   at Avalonia.NameGenerator.AvaloniaNameSourceGenerator.Execute(GeneratorExecutionContext context)
```

This is a limitation of [Microsoft.CodeAnalysis.GeneratorExecutionContext.AddSource](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.generatorexecutioncontext.addsource) hit by this code:

https://github.com/AvaloniaUI/Avalonia.NameGenerator/blob/cb79cd8f8fc8098506967a8f8d1d61f97ec76840/src/Avalonia.NameGenerator/Generator/AvaloniaNameGenerator.cs#L43-L48

Would it be possible to include the full namespace in the generated source file name, ie `FooPage.View.g.cs` vs `BarPage.View.g.cs`? This should guarantee uniqueness.

**Comment by worldbeater** [comment]
Thanks for reporting. Going to :ship: [the fix](https://github.com/AvaloniaUI/Avalonia.NameGenerator/pull/93) soon, hope our CD pipeline is still alive.

**Comment by worldbeater** [comment]
[Shipped as 1.4.1](https://www.nuget.org/packages/XamlNameReferenceGenerator/1.4.1).
